src/Compilers/Core/Portable/CommandLine/ErrorLogger.cs

```diff
@@ -45,91 +48,102 @@ public ErrorLogger(Stream stream, string toolName, string toolFileVersion, Versi
 
         private void WriteToolInfo(string name, string fileVersion, Version assemblyVersion)
         {
-            _writer.WriteObjectStart("toolInfo");
+            _writer.WriteObjectStart("tool");
             _writer.Write("name", name);
             _writer.Write("version", assemblyVersion.ToString(fieldCount: 3));
             _writer.Write("fileVersion", fileVersion);
             _writer.WriteObjectEnd();
         }
 
-        internal void LogDiagnostic(Diagnostic diagnostic, CultureInfo culture)
+        public void LogDiagnostic(Diagnostic diagnostic)
         {
             _writer.WriteObjectStart(); // result
             _writer.Write("ruleId", diagnostic.Id);
-            _writer.Write("kind", GetKind(diagnostic.Severity));
 
-            WriteLocations(diagnostic.Location, diagnostic.AdditionalLocations);
-
-            string message = diagnostic.GetMessage(culture);
-            if (string.IsNullOrEmpty(message))
+            string ruleIdKey = _descriptors.Add(diagnostic.Descriptor);
+            if (ruleIdKey != diagnostic.Id)
             {
-                message = "<None>";
+                _writer.Write("ruleIdKey", ruleIdKey);
             }
 
-            string description = diagnostic.Descriptor.Description.ToString(culture);
-            if (string.IsNullOrEmpty(description))
+            _writer.Write("level", GetLevel(diagnostic.Severity));
+
+            string message = diagnostic.GetMessage(_culture);
+            if (!string.IsNullOrEmpty(message))
             {
-                _writer.Write("fullMessage", message);
+                _writer.Write("message", message);
             }
-            else
+
+            if (diagnostic.IsSuppressed)
             {
-                _writer.Write("shortMessage", message);
-                _writer.Write("fullMessage", description);
+                _writer.WriteArrayStart("suppressionStates");
+                _writer.Write("suppressedInSource");
```

Comment:
Oh! Red alert! Two things:

1. @michaelcfanning: The schema correctly shows the values of the SuppressionStates enumeration as "suppressed", "suppressedInSource". But the code gen hints file gives the enumeration constant names as "SuppressedInSource", "SuppressedInBaseline", so the generated file SuppressionStates.cs is wrong. I'll fix that.

2. @nguerrera: The interpretation of the SuppressionStates enum is subtle. Every result that is suppressed, for whatever reason should have the value "suppressed" in the "suppressionStates" array. Those results that are suppressed because of an in-source suppression should *also* have the value "suppressedInSource" in the array. I can explain offline the reasoning behind that.

So, since in Roslyn every suppression you see is a source-level suppression (that's right, isn't it?) your code should be

                _writer.WriteArrayStart("suppressionStates");
                _writer.Write("suppressed");
                _writer.Write("suppressedInSource");

... resulting in

                "suppressionStates": [ "suppressed", "suppressedInSource" ]

---
In reply to: [63035326](https://github.com/dotnet/roslyn/pull/11233#discussion_r63035326) [](ancestors = 63035326)